### PR TITLE
Force shutdown websocket on application quit to prevent application from hanging

### PR DESCRIPTION
**Pull request by Hexalotl**
On standalone builds (such as Windows and Mac) there was a timing issue where the task used to cleanup the websockets was not updated if the application was about to quit. If we detect that the application will quit we instead choose to force quit the application.

### CODE REVIEW

File: Runtime/WebSocketConnection.cs [hunk 1]
```diff
@@ -134,17 +133,26 @@ public bool TryRemoveIncomingMessage(out byte[] message)
...
-            _cts.Cancel();
-            
             if (State == WebSocketState.Connecting || State == WebSocketState.Connected)
                 ChangeState(WebSocketState.Disconnecting);
         }
+
+        private void OnApplicationQuit()
+        {
+            if (State != WebSocketState.Connecting && State != WebSocketState.Connected)
+            {
```
Comment (mikerochip) [inline Runtime/WebSocketConnection.cs:148]: nit: delete brace

File: Runtime/WebSocketConnection.cs [hunk 2]
```diff
@@ -134,17 +133,26 @@ public bool TryRemoveIncomingMessage(out byte[] message)
...
+                return;
+            }
```
Comment (mikerochip) [inline Runtime/WebSocketConnection.cs:150]: nit: delete brace

**Review by mikerochip — CHANGES_REQUESTED** [review]
Thank you for this! This is also making me realize I should probably create a style guide as well as an `.editorconfig`.

File: Runtime/WebSocketConnection.cs (hunk 2 above)
Comment (Hexalotl) [reply Runtime/WebSocketConnection.cs]: Braces removed

**Comment by mikerochip** [comment]
Also just realized that using `destroyCancellationToken` would bump the min Unity version requirement up to 2022.2, so let's (unfortunately) remove that one

**Comment by Hexalotl** [comment]
`destroyCancellationToken` has been removed from the PR and now just captures the force shutdown step on application quit 👍

**Review by mikerochip — APPROVED** [review]
tyvm!

**Pull request merged**